src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from copy import deepcopy
+from argparse import Namespace
+import numpy as np
+import torch
+from tensorboardX import SummaryWriter
+
+from nni.compression.torch.compressor import Pruner
+from .channel_pruning_env import ChannelPruningEnv
+from .lib.agent import DDPG
+from .lib.utils import get_output_folder
+
+torch.backends.cudnn.deterministic = True
+
+class AMCPruner(Pruner):
+    """
+    A pytorch implementation of AMC: AutoML for Model Compression and Acceleration on Mobile Devices.
+    (https://arxiv.org/pdf/1802.03494.pdf)
+
+    Parameters:
+        model: nn.Module
+            The model to be pruned.
+        config_list: list
+            Configuration list to configure layer pruning.
+            Supported keys:
+                op_types: operation type to be pruned
+                op_names: operation name to be pruned
+        evaluator: function
+            function to evaluate the pruned model.
+            The prototype of the function:
+                >>> def val_func(val_loader, model):
+                >>>     ...
+                >>>     return acc
+        val_loader: torch.utils.data.DataLoader
+            Data loader of validation dataset.
+        suffix: str
+            suffix to help you remember what experiment you ran
+        export: boolean
+            False: search best pruned model and export after search.
+            True: export a searched model, before exporting model, you must run this pruner
```

Comment:
i am confused about `True` and `False`, could you explain a little more?

Reply:
Thanks,  export actually means export only.  Will update doc.
False:  search (train) + export
True:  export only based on previous search result.

Reply:
how about using `run_mode`, it value could be `search_export` or `export_only`, by default, it is `search_export`.

Reply:
for export only, if there is no previous search result, do we output warning?

Reply:
The name of this parameter previously was 'job':
 'train':  train (search) only 
'export':  export only

I changed the default behavior from 'train' to 'train' + 'export',
How about change it back to:
`job`:
`train_export`:  train and export after train
`export`: export only